frontend/src/metabase/dashboard/selectors.ts

```diff
@@ -183,9 +183,23 @@ export function getDashCardBeforeEditing(state: State, dashcardId: DashCardId) {
 export const getLoadingDashCards = (state: State) =>
   state.dashboard.loadingDashCards;
 
-export const getDashboardById = (state: State, dashboardId: DashboardId) => {
+export const getDashboardById = (
+  state: State,
+  dashboardId: DashboardId,
+): StoreDashboard => {
   const dashboards = getDashboards(state);
-  return dashboards[dashboardId];
+
+  const isEntityId = typeof dashboardId === "string";
+  if (isEntityId) {
+    return isNotNull(
+      Object.values(dashboards).find(
```

Comment:
Use a similar code we used in `findDashboardById` function in `enterprise/frontend/src/embedding-sdk-bundle/components/public/dashboard/use-common-dashboard-params.tsx`

Without this fix, the back button wouldn't show up when drilling.

<img width="1227" height="658" alt="Image" src="https://github.com/user-attachments/assets/62b83aa7-0559-4b44-814a-1c2e1a8ad141" />